aspnetcore/mvc/controllers/filters.md

```diff
@@ -178,246 +177,341 @@ In the following code, both the `ShortCircuitingResourceFilter` and the `AddHead
 
 The <xref:Microsoft.AspNetCore.Mvc.Filters.ActionExecutingContext> provides the following properties:
 
-* `ActionArguments` - lets you manipulate the inputs to the action.
-* `Controller` - lets you manipulate the controller instance. 
-* `Result` - setting this short-circuits execution of the action method and subsequent action filters. Throwing an exception also prevents execution of the action method and subsequent filters, but is treated as a failure instead of a successful result.
+* <xref:Microsoft.AspNetCore.Mvc.Filters.ActionExecutingContext.ActionArguments> - enables the inputs to an action method be read.
+* <xref:Microsoft.AspNetCore.Mvc.Controller> - enables manipulating the controller instance.
+* <xref:System.Web.Mvc.ActionExecutingContext.Result> - setting `Result` short-circuits execution of the action method and subsequent action filters.
+
+Throwing an exception in an action method:
+
+* Prevents running of subsequent filters.
+* Unlike setting `Result`, is treated as a failure instead of a successful result.
 
 The <xref:Microsoft.AspNetCore.Mvc.Filters.ActionExecutedContext> provides `Controller` and `Result` plus the following properties:
 
-* `Canceled` - will be true if the action execution was short-circuited by another filter.
-* `Exception` - will be non-null if the action or a subsequent action filter threw an exception. Setting this property to null effectively 'handles' an exception, and `Result` will be executed as if it were returned from the action method normally.
+* <xref:System.Web.Mvc.ActionExecutedContext.Canceled> - True if the action execution was short-circuited by another filter.
+* <xref:System.Web.Mvc.ActionExecutedContext.Exception> - Non-null if the action or a previously run action filter threw an exception. Setting this property to null:
 
-For an `IAsyncActionFilter`, a call to the `ActionExecutionDelegate`:
+  * Effectively handles the exception.
+  * `Result` is executed as if it was returned from the action method.
+
+For an `IAsyncActionFilter`, a call to the <xref:Microsoft.AspNetCore.Mvc.Filters.ActionExecutionDelegate>:
 
 * Executes any subsequent action filters and the action method.
-* returns `ActionExecutedContext`. 
+* Returns `ActionExecutedContext`.
 
-To short-circuit, assign `ActionExecutingContext.Result` to some result instance and don't call the `ActionExecutionDelegate`.
+To short-circuit, assign <xref:Microsoft.AspNetCore.Mvc.Filters.ActionExecutingContext.Result?displayProperty=fullName> to a result instance and don't call the `ActionExecutionDelegate`.
 
-The framework provides an abstract `ActionFilterAttribute` that you can subclass. 
+The framework provides an abstract <xref:Microsoft.AspNetCore.Mvc.Filters.ActionFilterAttribute> that can be subclassed.
 
-You can use an action filter to validate model state and return any errors if the state is invalid:
+The `OnActionExecuted` action filter can be used to:
```

Comment:
It would be more appropriate to use `OnActionExecuting` for this, as it runs before the model state is used rather than after. The statement is referring to a method but calling it an action filter, which is a little bit off. Perhaps the following would work:

> The 'OnActionExecuting` method can be used to: